tests/manifold/spherical_manifold_07.cc

```diff
@@ -0,0 +1,141 @@
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+// Check SphericalManifold on faces.
+
+#include "../tests.h"
+
+#include <deal.II/base/utilities.h>
+#include <deal.II/base/qprojector.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/fe/mapping_q_generic.h>
+#include <deal.II/fe/mapping_manifold.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/grid/manifold_lib.h>
+
+// Check that the spherical manifold finds the right intermediate
+// points both using face quadratures, as well as using face
+// quadratures projected to cells.
+
+#include <numeric>
+
+template<int dim, int spacedim>
+void test()
+{
+  deallog << "dim=" << dim << ", spacedim=" << spacedim << std::endl;
+
+  Triangulation<dim, spacedim>   triangulation;
+
+  GridGenerator::hyper_cube (triangulation, 2.0, 4.0);
+
+  typename Triangulation<dim,spacedim>::active_cell_iterator
+  cell = triangulation.begin_active();
+
+  // Center and radius of the Ball
+  Point<spacedim> center = cell->center();
+  double radius = center.distance(cell->vertex(0));
+
+  static  const SphericalManifold<dim,spacedim> manifold(cell->center());
+
+  triangulation.set_all_manifold_ids(0);
+  triangulation.set_manifold (0, manifold);
+
+  const QGauss<dim-1> face_quad(3);
+
+  const FE_Q<dim-1> face_fe_q(1);
+
+  // Compute the points with the faces
+  deallog << "Face quadratures ("
+	  << face_quad.size() * GeometryInfo<dim>::faces_per_cell
+	  << ")" << std::endl;
+  
+  for(unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f) {
+    std::vector<Point<spacedim> > vertices;
+    std::vector<double> weights(GeometryInfo<dim>::vertices_per_face);
+    
+    for(unsigned int i=0; i<GeometryInfo<dim>::vertices_per_face; ++i)
+      vertices.push_back(cell->face(f)->vertex(i));
+    
+    for (unsigned int i=0; i<face_quad.size(); ++i)
+    {
+      for(unsigned int v=0; v<weights.size(); ++v)
+	weights[v] = face_fe_q.shape_value(v, face_quad.point(i));
+
+      deallog << manifold.get_new_point(Quadrature<spacedim>(vertices, weights))
+	      << std::endl;
+    }
+  }
+
+  // Project the face quads
+  Quadrature<dim> quad = QProjector<dim>::project_to_all_faces(face_quad);
+
+  deallog << "Face quadratures projected on cell "
+	  << "(" << quad.size() << ")" << std::endl;
+
+  std::vector<Point<spacedim> > vertices;
+  for(unsigned int i=0; i<GeometryInfo<dim>::vertices_per_cell; ++i) 
+    vertices.push_back(cell->vertex(i));
+
+  // Used to compute weights.
+  FE_Q<dim> fe_q(1);
+
+  std::vector<double> weights(GeometryInfo<dim>::vertices_per_cell);
+  
+  for (unsigned int i=0; i<quad.size(); ++i)
+    {
+      for(unsigned int v=0; v<weights.size(); ++v)
+	weights[v] = fe_q.shape_value(v, quad.point(i));
+
+      deallog << manifold.get_new_point(Quadrature<spacedim>(vertices, weights))
+	      << std::endl;
+    }
```

Comment:
So I'm not sure what exactly you do here. You loop over all 12 points along the perimeter of the reference cell, then compute for each of them the 4 weights, and compute the new-point operation for each of the points with the 4 vertices and 4 weights. I would expect this to yield 12 points on the perimeter of the *mapped* cell. Is this what you would expect as well?